src/runtime/GlobalObjectBuiltinPromise.cpp

```diff
@@ -327,13 +328,39 @@ static Value builtinPromiseCatch(ExecutionState& state, Value thisValue, size_t
 }
 
 
+static Value builtinPromiseFinally(ExecutionState& state, Value thisValue, size_t argc, Value* argv, bool isNewExpression)
+{
+    auto strings = &state.context()->staticStrings();
+    if (!thisValue.isObject())
+        ErrorObject::throwBuiltinError(state, ErrorObject::TypeError, strings->Promise.string(), false, strings->stringFinally.string(), "%s: not a Promise object");
+    Object* thisObject = thisValue.asObject();
+    Value onFinally = argv[0];
+    ExtendedNativeFunctionObject* thenFinally;
+    ExtendedNativeFunctionObject* catchFinally;
+    Value arguments[] = { onFinally, onFinally };
+    Value C = thisObject->speciesConstructor(state, state.context()->globalObject()->promise());
+    if (onFinally.isCallable()) {
+        SmallValueVector valueArgs;
+        valueArgs.push_back(C);
+        valueArgs.push_back(onFinally);
+        thenFinally = new ExtendedNativeFunctionObject(state, NativeFunctionInfo(AtomicString(state, String::emptyString), promiseThenFinally, 1, NativeFunctionInfo::Strict), valueArgs);
+        catchFinally = new ExtendedNativeFunctionObject(state, NativeFunctionInfo(AtomicString(state, String::emptyString), promiseCatchFinally, 1, NativeFunctionInfo::Strict), valueArgs);
```

Comment:
This `valueArgs` vector is temporally allocated, so the move constructor of Vector could be used to initialize `ExtendedNativeFunctionObject` by calling `std::move` here.